tests/integration/cairo1_hint_locals/deploy/deploy_test.cairo

```diff
@@ -0,0 +1,45 @@
+use array::ArrayTrait;
+use result::ResultTrait;
+
+#[test]
+fn test_deploy() {
+    let class_hash = declare('minimal').unwrap();
+    assert(class_hash != 0, 'class_hash != 0');
+
+    let prepare_result = prepare(class_hash, ArrayTrait::new()).unwrap();
+
+    assert(prepare_result.contract_address != 0, 'deployed contract_address != 0');
+    assert(prepare_result.class_hash != 0, 'deployed class_hash != 0');
+
+    let prepared_contract = PreparedContract {
+        contract_address: prepare_result.contract_address,
+        class_hash: prepare_result.class_hash,
+        constructor_calldata: prepare_result.constructor_calldata
+    };
+    deploy(prepared_contract).unwrap();
+}
+
+#[test]
+fn test_deploy_with_ctor() {
+    let mut constructor_calldata = ArrayTrait::new();
+    constructor_calldata.append(1);
+    constructor_calldata.append(2);
+
+
+    let class_hash = declare('with_ctor').unwrap();
+    assert(class_hash != 0, 'declared class_hash != 0');
+
+    let prepare_result = prepare(class_hash, constructor_calldata).unwrap();
+
+    assert(prepare_result.contract_address != 0, 'deployed contract_address != 0');
+    assert(prepare_result.class_hash != 0, 'deployed class_hash != 0');
+    // TODO: check the length of the array, this produces: error: Variable was previously moved
+    // assert(prepare_result.constructor_calldata.len() == 2_u32, 'constructor_calldata size == 2');
+
+    let prepared_contract = PreparedContract {
+        contract_address: prepare_result.contract_address,
+        class_hash: prepare_result.class_hash,
+        constructor_calldata: prepare_result.constructor_calldata
+    };
+    deploy(prepared_contract).unwrap();
```

Comment:
We probably should handle the scenario when contract panics in constructor with error `'example error content'`
And we can do an assert of the error message in tests
```
match deploy(prepared) {
   Err('example error content') => {}
   _ => panic()
}
```

Reply:
[ad0d8ed](https://github.com/software-mansion/protostar/pull/1716/commits/ad0d8ed8c14630f7b36e0aa9dce937673b3d36b5) is this ok?
I just passed wrong number of args there, not sure if it's necessary to write yet another contract with a static `panic` in the constructor?

Reply:
There is, we want to test an error raised from constructor and assert the error message.

Invalid number of arguments is also a good test but those are testing different functionalities

Reply:
[add panicking ctor test](https://github.com/software-mansion/protostar/pull/1716/commits/ec2a6c00c7ed227626d63603e9e9d202d9c863c2)